components/src/components/top-button/top-button.tsx

```diff
@@ -0,0 +1,31 @@
+import { Component, Listen, Prop, h } from '@stencil/core';
+
+// Scroll to top button.
+@Component({
+  tag: 'pulumi-top-button',
+  styleUrl: 'top-button.scss',
+  shadow: false
+})
+
+export class TopButton {
+
+  @Prop({ mutable: true })
+  scrollPostion: number;
+
+  @Listen('scroll', { target: 'window' })
+  handleScroll() {
+    this.scrollPostion = window.scrollY;
+  }
+
+  componentWillRender() {
+    this.scrollPostion = window.scrollY;
+  }
+
+  render() {
+    return (
+      <div>
+          <a class="top-button fas fa-chevron-up" title="Scroll to top" style={{display: this.scrollPostion < 2500 ? "none" : ""}}></a>
```

Comment:
Rather than explicitly set a style within the code of the component, you might consider applying a CSS class instead, and put the definition of that class in the Sass along with the other visual things.